src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```diff
@@ -0,0 +1,1202 @@
+    protected WorkflowDocumentService workflowDocumentService;
+
+    protected String attachmentsDirectory;
+    private int achTotalPostedTransactions;
+    private int achTotalSkippedTransactions;
+    private int wiredTotalPostedTransactions;
+    private int wiredTotalSkippedTransactions;
+    private KualiDecimal achTotalPostedTransactionAmount;
+    private KualiDecimal achTotalSkippedTransactionAmount;
+    private KualiDecimal wiredTotalPostedTransactionAmount;
+    private KualiDecimal wiredTotalSkippedTransactionAmount;
+
+    /**
+     * @see AdvanceDepositService#createDocuments()
+     */
+    public void createDocuments() {
+        List<AchIncomeTransaction> transactions = (List<AchIncomeTransaction>) businessObjectService.findMatchingOrderBy(AchIncomeTransaction.class, new HashMap(), ACH_TRANSACTION_SEQUENCE_NUMBER, true);
+
+        for (AchIncomeTransaction transaction : transactions) {
+            AdvanceDepositDocument document = createAdvanceDepositDocument(transaction);
+            saveDocument(document);
+            createNotes(transaction, document);
+        }
+    }
+
+    private void saveDocument(AdvanceDepositDocument document) {
+        try {
+            documentService.saveDocument(document);
+            if (LOG.isInfoEnabled()) {
+                LOG.info("Saved Advance Deposit document: " + document.getDocumentNumber());
+            }
+        } catch (Exception e) {
+            LOG.error("Error persisting document # " + document.getDocumentHeader().getDocumentNumber() + " " + e.getMessage(), e);
+            throw new RuntimeException("Error persisting document # " + document.getDocumentHeader().getDocumentNumber() + " " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * @see AdvanceDepositService#routeAdvanceDepositDocuments()
+     */
+    public boolean routeAdvanceDepositDocuments() {
+        List<String> documentIdList;
+
+        try {
+            documentIdList = retrieveAdvanceDepositDocumentsToRoute(KewApiConstants.ROUTE_HEADER_SAVED_CD);
+        } catch (WorkflowException | RemoteException e) {
+            LOG.error("Error retrieving advance deposit documents for routing: " + e.getMessage(), e);
+            throw new RuntimeException(e.getMessage(), e);
+        }
+
+        if (LOG.isInfoEnabled()) {
+            LOG.info("Advance deposit to Route: " + documentIdList);
+        }
+
+        for (String documentId : documentIdList) {
+            try {
+                AdvanceDepositDocument advanceDocument = (AdvanceDepositDocument) documentService.getByDocumentHeaderId(documentId);
+                if (LOG.isInfoEnabled()) {
+                    LOG.info("Routing advance deposit document # " + documentId + ".");
+                }
+                documentService.prepareWorkflowDocument(advanceDocument);
+                workflowDocumentService.route(advanceDocument.getDocumentHeader().getWorkflowDocument(), "document routed by achIncome batch job", null);
```

Comment:
The DocumentService routeDocument validates and persists the document as well as calling WorkflowDocumentService.route(). Not sure if the code was intentionally avoiding those calls. I'll try it with the DocumentService.route and see what happens.